### PR TITLE
Display reloaded values of server settings

### PR DESCRIPTION
**Pull request by helifu**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Display the correct server settings after reload.

### CODE REVIEW

**Comment by pufit** [comment]
Hi @helifu! Can you merge master into your branch?

**Comment by helifu** [comment]
> Hi @helifu! Can you merge master into your branch?

done.

File: src/Storages/System/StorageSystemServerSettings.cpp [hunk 1]
```diff
@@ -21,15 +33,40 @@ NamesAndTypesList StorageSystemServerSettings::getNamesAndTypes()
...
+        {"background_fetches_pool_size", std::to_string(context->getFetchesExecutor()->getMaxThreads())},
+        {"background_common_pool_size", std::to_string(context->getCommonExecutor()->getMaxThreads())},
+
+        {"background_buffer_flush_schedule_pool_size", std::to_string(CurrentMetrics::get(CurrentMetrics::BackgroundBufferFlushSchedulePoolSize))},
+        {"background_schedule_pool_size", std::to_string(CurrentMetrics::get(CurrentMetrics::BackgroundSchedulePoolSize))},
+        {"background_message_broker_schedule_pool_size", std::to_string(CurrentMetrics::get(CurrentMetrics::BackgroundMessageBrokerSchedulePoolSize))},
+        {"background_distributed_schedule_pool_size", std::to_string(CurrentMetrics::get(CurrentMetrics::BackgroundDistributedSchedulePoolSize))}
+    };
+
     const auto & config = context->getConfigRef();
```
Comment (pufit) [inline src/Storages/System/StorageSystemServerSettings.cpp:59]: The problem is that, for some reason, `context->getConfigRef()` returns an old version of the config after reloading. If we can fix this, `system.server_settings` table will return reloaded values automatically with no additional code required.

**Review by pufit — CHANGES_REQUESTED** [review]

File: src/Storages/System/StorageSystemServerSettings.cpp (hunk 1 above)
Comment (helifu) [reply src/Storages/System/StorageSystemServerSettings.cpp:59]: The reason that I added these additinal code is that some configurations do not take effect even after reloading.
Look at the [main_config_reloader](https://github.com/ClickHouse/ClickHouse/blob/8a1363bcf1124498fd46f68e2aa671e5175bf321/programs/server/Server.cpp#L1169C8-L1169C8), we should not display the configurations that do not work.

Comment (pufit) [reply src/Storages/System/StorageSystemServerSettings.cpp:59]: ok, you are right. Can you also add a new column `is_hot_reloadable`, which will be `true` for all configs in your list?

Comment (helifu) [reply src/Storages/System/StorageSystemServerSettings.cpp:60]: Done.

**Review by pufit — APPROVED** [review]

**Pull request merged**